Improve ark documentation for backups and restore (#2365)
kyma-project/kyma: tools/ark-plugins/README.md

```bash
ark plugin add {yourRepo/imageName:tag}
```

> ## Add a plugin

Reviewer: `Add a plugin` -> `Create a new plugin` ?

Author: I changed it to `Create a new plugin`

- New plugins must be added under [`internal/plugins`](internal/plugins) :

```             
  ├── internal                                                                  
    ├── plugins

>       ├── backup    # new Backup Item Action plugins 

Reviewer: for now there's no subfolders in plugins, and for now I don't see any usecase for backup actions - do you?

Author: I was thinking that in the future other teams might want to add their owns plugins and having these folders may provides a way of organize them. 

Originally I had `internal/plugins/backup/function.go` and  `internal/plugins/restore/function.go`. However I did not include them in this PR because of the issue with `dep status -v`.

Reviewer: sure, will you also move existing plugins? it may be in followup PR

      ├── restore   # new Restore Item Action plugins 
  ```

> - All new plugin needs to be registered in the file `tools/ark-plugins/main.go`

Reviewer 2: ```suggestion
- All new plugins must be registered in the `tools/ark-plugins/main.go` file.
```

> - **Backup Item Action** - performs arbitrary logic on individual items prior to storing them in the backup file.

Reviewer: same as above, I'm not against adding subfolders but I don't see any backup actions yet

Author: I understand but we can't tell that **`BIA`** won"t be used. Actually part of the issue we had when the backup of lambdas is executed might solve with **`BIA`**.

This error is present after executing the restore.

```Events:
  Type     Reason                               Age   From                                Message
  ----     ------                               ----  ----                                -------
  Warning  ProvisionCallFailed                  5m    service-catalog-controller-manager  Error provisioning ServiceInstance of ClusterServiceClass (K8S: "a2257daa-0e26-4c61-a68d-8a7453c1b767" ExternalName: "redis") at ClusterServiceBroker "core-helm-broker": Status: 400; ErrorMessage: while inserting instance to storage: element already exists; Description: ; ResponseError: <nil>
  Warning  ClusterServiceBrokerReturnedFailure  5m    service-catalog-controller-manager  Error provisioning ServiceInstance of ClusterServiceClass (K8S: "a2257daa-0e26-4c61-a68d-8a7453c1b767" ExternalName: "redis") at ClusterServiceBroker "core-helm-broker": Status: 400; ErrorMessage: while inserting instance to storage: element already exists; Description: ; ResponseError: <nil>
```

Reviewer: ok, the same as above

- **Restore Item Action** - performs arbitrary logic on individual items prior to restoring them in the Kyma cluster.

> - All plugin must be implemented as the following example:

Reviewer 2: ```suggestion
- All plugins must be implemented as the following example:
```
